common/src/main/java/com/microsoft/identity/common/internal/dto/PrimaryRefreshTokenRecord.java

```diff
@@ -0,0 +1,252 @@
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.dto;
+
+import com.google.gson.annotations.SerializedName;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
+
+import static com.microsoft.identity.common.internal.dto.PrimaryRefreshTokenRecord.SerializedNames.*;
+
+public class PrimaryRefreshTokenRecord extends Credential {
+
+    public static class SerializedNames extends Credential.SerializedNames {
+        /**
+         * String of family id.
+         */
+        public static final String FAMILY_ID = "family_id";
+
+        /**
+         * String of session_key.
+         */
+        public static final String SESSION_KEY = "session_key";
+
+        /**
+         * String of session_key_rolling_date.
+         */
+        public static final String SESSION_KEY_ROLLING_DATE = "session_key_rolling_date";
+
+        /**
+         * String of prt_protocol_version.
+         */
+        public static final String PRT_PROTOCOL_VERSION = "prt_protocol_version";
+    }
+
+    /**
+     * 1st Party Application Family ID.
+     */
+    @SerializedName(FAMILY_ID)
+    private String mFamilyId;
+
+    /**
+     * PRT expiry time. This value is returned from the server as refresh_token_expires_in that
+     * should be added to local time.
+     */
+    @SerializedName(EXPIRES_ON)
+    private String mExpiresOn;
+
+    /**
+     * Session key of PRT.
+     */
+    @SerializedName(SESSION_KEY)
+    private String mSessionKey;
+
+    /**
+     * The version of the PRT Protocol being used, version 3 supports no device_id.
+     */
+    @SerializedName(PRT_PROTOCOL_VERSION)
+    private String mPrtProtocolVersion;
+
+    /**
+     * Session key expiry time. This value is determined by the client and set to 60 days after the
+     * session key was initally issued
+     */
+    @SerializedName(SESSION_KEY_ROLLING_DATE)
+    private String mSessionKeyRollingDate;
+
+    /**
+     * Gets the family_id.
+     *
+     * @return The family_id to get.
+     */
+    public String getFamilyId() {
+        return mFamilyId;
+    }
+
+    /**
+     * Sets the family_id.
+     *
+     * @param familyId The family_id to set.
+     */
+    public void setFamilyId(final String familyId) {
+        mFamilyId = familyId;
+    }
+
+    public boolean isExpired(final String expires) {
+        // Init a Calendar for the current time/date
+        final Calendar calendar = Calendar.getInstance();
+        final Date validity = calendar.getTime();
+        // Init a Date for the accessToken's expiry
+        long epoch = Long.valueOf(expires);
+        final Date expiresOn = new Date(
+                TimeUnit.SECONDS.toMillis(epoch)
+        );
+        return expiresOn.before(validity);
```

Comment:
Could this just be Long.valueOf(expires) <  System.currentTimeMillis?

Reply:
Updated, thanks